---
title: "portsmouth-darek-stemcells"
output:
  html_document:
    toc: yes
    df_print: paged

date: "_Ostatnio sporządzony: `r format(Sys.time(), '%d %B, %Y %H:%M:%S')`_"

knit: (function(input, ...) {
    rmarkdown::render(
      input,
      output_file = paste0(
        xfun::sans_ext(input), '-tmp.html'
      ),
      envir = globalenv()
    )
  })
---

<STYLE TYPE="text/css">
  table {
    font-size: 12px;
  }
</STYLE>

```{r setup, include=FALSE}
library(knitr)
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
```

```{bash download_r_package, eval=FALSE}
# R-script: RNA-seq Analysis of HSC-MPPs
# from https://doi.org/10.1016/j.stem.2014.07.005
wget "https://www.cell.com/cms/10.1016/j.stem.2014.07.005/attachment/4b75e9f1-97f4-4d53-beba-7e4e023f5969/mmc8.zip" \
    -P raw/package/
unzip raw/package/mmc8.zip -d raw/package/
tar -C raw/package/ -czf raw/package/mouse.RNAseq.HSC.MPP.tar.gz mouse.RNAseq.HSC.MPP/
```


```{r install_packages, eval=FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("DEXSeq")

install.packages(
  "raw/package/mouse.RNAseq.HSC.MPP.tar.gz",
  repos = NULL,
  type = "source"
)


install.packages("ggpubr")
```

```{r load_libraries, message=FALSE}
library(kableExtra)
library(ggplot2)
library(ggpubr)
library(mouse.RNAseq.HSC.MPP)
library(tidyverse)
```

```{r}
data("dseHSCMPP")
data("ecsHSCMPP")
data("ecsHSCMPP1")

DMD = "ENSMUSG00000045103"
```


# dseHSCMPP

```{r eval=FALSE}
setClass("DataFrame", representation(nrows = "numeric", age = "numeric"))
dummy_slot <- new("DataFrame", nrows = 0, age = 31)
dseHSCMPP@elementMetadata <- dummy_slot
str(dseHSCMPP)
```

```{r}
slotNames(dseHSCMPP)
str(dseHSCMPP@assays$data$counts)
dseHSCMPP@assays$data$counts %>%
  as_tibble(rownames = 'gene') %>%
  filter(gene == DMD)
```


# ecsHSCMPP

```{r}
str(ecsHSCMPP)
```

```{r}
str(ecsHSCMPP@assayData)
ecsHSCMPP@assayData$counts %>%
  as.tibble(rownames = 'gene:exon') %>%
  filter(substr(`gene:exon`, start = 1, stop = 18) == DMD)
```

```{r}
str(ecsHSCMPP@featureData)
ecsHSCMPP@featureData@varMetadata
ecsHSCMPP@featureData@data %>%
  as_tibble(rownames = 'gene:exon') %>%
  filter(geneID == DMD)
```


# ecsHSCMPP1

Table poniżej są tymi samymi co z _ecsHSCMPP_, ale tylko dla komórek HSC i MPP1.

```{r}
x <- ecsHSCMPP@assayData$counts %>%
  as.tibble(rownames = 'gene:exon') %>%
  filter(substr(`gene:exon`, start = 1, stop = 18) == DMD) %>%
  select(`gene:exon`, contains('HSC'), contains('MPP1'))

y <- ecsHSCMPP1@assayData$counts %>%
  as.tibble(rownames = 'gene:exon') %>%
  filter(substr(`gene:exon`, start = 1, stop = 18) == DMD) %>%
  select(`gene:exon`, contains('HSC'), contains('MPP1'))

stopifnot(identical(x, y))
```

```{r}
str(ecsHSCMPP1)
```

```{r}
str(ecsHSCMPP1@assayData)
ls(ecsHSCMPP1@assayData)
ecsHSCMPP1@assayData$counts %>%
  as.tibble(rownames = 'gene:exon') %>%
  filter(substr(`gene:exon`, start = 1, stop = 18) == DMD)
```

```{r}
str(ecsHSCMPP1@featureData)
ecsHSCMPP1@featureData@varMetadata
ecsHSCMPP1@featureData@data %>%
  as_tibble(rownames = 'gene:exon') %>%
  filter(geneID == DMD)
```
