---
title: "portsmouth-darek-stemcells"
output:
  html_document:
    toc: yes
    df_print: paged

date: "_Ostatnio sporządzony: `r format(Sys.time(), '%d %B, %Y %H:%M:%S')`_"

knit: (function(input, ...) {
    rmarkdown::render(
      input,
      output_file = paste0(
        xfun::sans_ext(input), '-tmp.html'
      ),
      envir = globalenv()
    )
  })
---

<STYLE TYPE="text/css">
  table {
    font-size: 12px;
  }
</STYLE>

```{r setup, include=FALSE}
library(knitr)
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
```

```{bash download_r_package, eval=FALSE}
# R-script: RNA-seq Analysis of HSC-MPPs
# from https://doi.org/10.1016/j.stem.2014.07.005
wget "https://www.cell.com/cms/10.1016/j.stem.2014.07.005/attachment/4b75e9f1-97f4-4d53-beba-7e4e023f5969/mmc8.zip" \
    -P raw/package/
unzip raw/package/mmc8.zip -d raw/package/
tar -C raw/package/ -czf raw/package/mouse.RNAseq.HSC.MPP.tar.gz mouse.RNAseq.HSC.MPP/
```


```{r install_packages, eval=FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("DEXSeq")

install.packages(
  "raw/package/mouse.RNAseq.HSC.MPP.tar.gz",
  repos = NULL,
  type = "source"
)


install.packages("ggpubr")
```

```{r load_libraries, message=FALSE}
library(kableExtra)
library(ggplot2)
library(ggpubr)
library(mouse.RNAseq.HSC.MPP)
library(tidyverse)
```

```{r}
data("dseHSCMPP")
data("ecsHSCMPP")
data("ecsHSCMPP1")

DMD = "ENSMUSG00000045103"
```


# dseHSCMPP

```{r eval=FALSE}
setClass("DataFrame", representation(nrows = "numeric", age = "numeric"))
dummy_slot <- new("DataFrame", nrows = 0, age = 31)
dseHSCMPP@elementMetadata <- dummy_slot
str(dseHSCMPP)
```

```{r}
slotNames(dseHSCMPP)
str(dseHSCMPP@assays$data$counts)
dseHSCMPP@assays$data$counts %>%
  as_tibble(rownames = 'gene') %>%
  filter(gene == DMD)
```


# ecsHSCMPP

```{r}
str(ecsHSCMPP)
```

```{r}
str(ecsHSCMPP@assayData)
ecsHSCMPP@assayData$counts %>%
  as.tibble(rownames = 'gene:exon') %>%
  filter(substr(`gene:exon`, start = 1, stop = 18) == DMD)
```

```{r}
str(ecsHSCMPP@featureData)
ecsHSCMPP@featureData@varMetadata
ecsHSCMPP@featureData@data %>%
  as_tibble(rownames = 'gene:exon') %>%
  filter(geneID == DMD)
```


# ecsHSCMPP1

Table poniżej są tymi samymi co z _ecsHSCMPP_, ale tylko dla komórek HSC i MPP1.

```{r}
x <- ecsHSCMPP@assayData$counts %>%
  as.tibble(rownames = 'gene:exon') %>%
  filter(substr(`gene:exon`, start = 1, stop = 18) == DMD) %>%
  select(`gene:exon`, contains('HSC'), contains('MPP1'))

y <- ecsHSCMPP1@assayData$counts %>%
  as.tibble(rownames = 'gene:exon') %>%
  filter(substr(`gene:exon`, start = 1, stop = 18) == DMD) %>%
  select(`gene:exon`, contains('HSC'), contains('MPP1'))

stopifnot(identical(x, y))
```

```{r}
str(ecsHSCMPP1)
```

```{r}
str(ecsHSCMPP1@assayData)
ls(ecsHSCMPP1@assayData)
ecsHSCMPP1@assayData$counts %>%
  as.tibble(rownames = 'gene:exon') %>%
  filter(substr(`gene:exon`, start = 1, stop = 18) == DMD)
```

```{r}
str(ecsHSCMPP1@featureData)
ecsHSCMPP1@featureData@varMetadata
ecsHSCMPP1@featureData@data %>%
  as_tibble(rownames = 'gene:exon') %>%
  filter(geneID == DMD)
```

# Expression of Dp72 and Dp427

```{r load_fpkm}
cuff_path <- file.path(
  'data',
  'rna-seq-paired-end-output',
  'glob-c589286a63ff6674632a91417e56ef8a'
)

isoforms_fpkm <- read_tsv(file.path(cuff_path, 'isoforms.fpkm_table.tsv'))
isoforms_attr <-
  read_tsv(file.path(cuff_path, 'isoforms.attr_table.tsv')) %>%
  select(tracking_id, gene_id, gene_short_name)

dmd_fpkm <-
    left_join(isoforms_fpkm, isoforms_attr, by = 'tracking_id') %>%
    filter(gene_short_name == 'Dmd')
```

```{r plot_figure}
ggdata <-
    dmd_fpkm %>%
    pivot_longer(
      c(starts_with('HSC'), starts_with('MPP')),
      names_to = 'cell_name',
      values_to = 'FPKM'
    ) %>%
    mutate(`cell type` = substr(cell_name, 1, nchar(cell_name) - 2)) %>%
    group_by(tracking_id, `cell type`) %>%
    summarise(`mean FPKM` = mean(FPKM), sem = sd(FPKM) / sqrt(n())) %>%
    mutate(
      `tracking id` = paste(
        tracking_id,
        if_else(tracking_id == 'ENSMUST00000114000', '(Dp427)', '(Dp71)')
      )
    )

ggplot(ggdata, aes(x = `cell type`, y = `mean FPKM`, fill = `tracking id`)) +
  geom_bar(stat = 'identity', position = 'dodge', show.legend = TRUE) +
  geom_errorbar(
    aes(ymin = `mean FPKM` - sem, ymax = `mean FPKM` + sem),
    position = position_dodge(width = .9),
    width = .3,
    size = 0.3
  ) +
  scale_fill_brewer(palette = 'Set1') +
  theme_pubr() +
  theme(legend.position = "right")
ggsave('analysis/FPKM_dmd.png', width = 8, height = 4, dpi = 150)
```

# Dp72 and Dp427 expression in comparison to the rest of transcripts

```{r load_fpkm2}
cuff_path <- file.path(
  'data',
  'rna-seq-paired-end-output',
  'glob-c589286a63ff6674632a91417e56ef8a'
)

isoforms_fpkm <- read_tsv(file.path(cuff_path, 'isoforms.fpkm_table.tsv'))
isoforms_attr <-
  read_tsv(file.path(cuff_path, 'isoforms.attr_table.tsv')) %>%
  select(tracking_id, gene_id, gene_short_name)
isoforms_data <- left_join(isoforms_fpkm, isoforms_attr, by = 'tracking_id')

genes_fpkm <- read_tsv(file.path(cuff_path, 'genes.fpkm_table.tsv'))
genes_attr <-
  read_tsv(file.path(cuff_path, 'genes.attr_table.tsv')) %>%
  select(tracking_id, gene_id, gene_short_name)
genes_data <- left_join(genes_fpkm, genes_attr, by = 'tracking_id')

hk_genes_data <-
    genes_data %>%
    filter(gene_short_name %in% c("Dmd", "Gapdh", "Hprt"))

hk_isoforms_data <-
    isoforms_data %>%
    filter(gene_short_name %in% c("Dmd", "Gapdh", "Hprt"))
```

```{r expression_level}
dp427_ens <- 'ENSMUST00000114000'
dp71_ens <- 'ENSMUST00000239019'

isoforms_data_agg <-
  isoforms_data %>%
  pivot_longer(c(starts_with("HSC"), starts_with("MPP")), names_to = "cell_type", values_to = "fpkm") %>%
  mutate(cell_type = substr(cell_type, 1, nchar(cell_type) - 2)) %>%
  group_by(gene_id, gene_short_name, tracking_id, cell_type) %>%
  summarise(mean_fpkm = mean(fpkm)) %>%
  ungroup() %>%
  pivot_wider(names_from = cell_type, values_from = mean_fpkm)
genes_data_agg <-
  genes_data %>%
  pivot_longer(c(starts_with("HSC"), starts_with("MPP")), names_to = "cell_type", values_to = "fpkm") %>%
  mutate(cell_type = substr(cell_type, 1, nchar(cell_type) - 2)) %>%
  group_by(gene_id, gene_short_name, tracking_id, cell_type) %>%
  summarise(mean_fpkm = mean(fpkm)) %>%
  ungroup() %>%
  pivot_wider(names_from = cell_type, values_from = mean_fpkm)

.names_map_df <- as_tibble(
  list(
    tracking_id = c(dp71_ens, dp427_ens),
    Name = c(paste(dp71_ens, "(dp71) rank"), paste(dp427_ens, "(dp427) rank"))
  )
)
dmd_ranks <-
  isoforms_data_agg %>%
  mutate(across(c(HSC, starts_with("MPP")), ~ min_rank(desc(.x)))) %>%
  filter(tracking_id %in% c(dp71_ens, dp427_ens)) %>%
  left_join(.names_map_df, by = 'tracking_id')

hprt <-
  genes_data_agg %>%
  filter(gene_short_name == "Hprt") %>%
  select(starts_with("HSC"), starts_with("MPP")) %>%
  as_vector()
gapdh <-
  genes_data_agg %>%
  filter(gene_short_name == "Gapdh") %>%
  select(starts_with("HSC"), starts_with("MPP")) %>%
  as_vector()
dp427_dmd <-
  isoforms_data_agg %>%
  filter(tracking_id == dp427_ens) %>%
  select(starts_with("HSC"), starts_with("MPP")) %>%
  as_vector()
dp71_dmd <-
  isoforms_data_agg %>%
  filter(tracking_id == dp71_ens) %>%
  select(starts_with("HSC"), starts_with("MPP")) %>%
  as_vector()

.n_isoforms <- as.list(rep(dim(isoforms_data)[1], 5))
names(.n_isoforms) <- names(select(dmd_ranks, HSC, starts_with("MPP")))
```

```{r expression_level_save}
dmd_ranks %>%
  bind_rows(c(.n_isoforms, Name = "n")) %>%
  bind_rows(c(as.list(round(dp427_dmd / hprt * 100, 2)), Name = "Dp427 / Hprt [%]")) %>%
  bind_rows(c(as.list(round(dp427_dmd / gapdh * 100, 2)), Name = "Dp427 / Gapdh [%]")) %>%
  bind_rows(c(as.list(round(dp71_dmd / hprt * 100, 2)), Name = "Dp71 / Hprt [%]")) %>%
  bind_rows(c(as.list(round(dp71_dmd / gapdh * 100, 2)), Name = "Dp71 / Gapdh [%]")) %>%
  select(-tracking_id, -gene_id, -gene_short_name) %>%
  relocate(Name) %>%
  write_csv('analysis/poziom-ekspresji-dmd.csv')
```

# Expression comparison between MPP1 and MPP3 / MPP4

```{r siginficant_MPP1_vs_MPP34}
library(rstatix)

dmd_fpkm_long <-
  dmd_fpkm %>%
  pivot_longer(c(contains("HSC"), contains("MPP")), names_to = "cell", values_to = "fpkm") %>%
  separate(cell, into = c("cell", "sample_nr")) %>%
  select(-gene_id, -gene_short_name)

dmd_fpkm_long %>%
  anova_test(
    dv = fpkm,
    between = c(cell, tracking_id)
  )

dmd_fpkm_long %>%
  filter(tracking_id == "ENSMUST00000239019") %>%
  pairwise_t_test(
    formula = fpkm ~ cell,
    p.adjust.method = "bonf"
  ) %>%
  arrange(group1)
```
